### PR TITLE
[WIP] feat(prettier): output to stdout instead of write file by default unless specified --write flag

### PR DESCRIPTION
**Pull request by axetroy**
### Why?

1. In order to stay consistent with `prettier cli`
2. for `deno fmt example.ts` to print formatted code in stdout
3. better integrated with IDE(eg. vscode/webstorm/atom...)

before: without `--write`

```bash
deno run prettier/main.ts script1.ts --allow-read --allow-write
```

after: with `--write`

```bash
deno run prettier/main.ts script1.ts --allow-read --allow-write --write
```

This PR bring **BREAKING CHANGE**: `Prettier` now output to stdout instead of overwriting the file by default. unless specified `--write` flag

### When should be output to stdout?

1. without `--write`/`--check` flag
2. when `source content` not equal to `formatted content`

### CODE REVIEW

**Comment by kt3k** [comment]
I think this feature is nice to have.

We need to be careful with the timing of merging of this PR because this will break `deno fmt` temporarily.

**Comment by axetroy** [comment]
@ry it's ready for review.

**Comment by ry** [comment]
@axetroy It LGTM. I'm a bit concerned that this will break "deno fmt" (we should be using a tagged link to std/prettier and not master link).

**Comment by axetroy** [comment]
@ry yes. it will break the `demo fmt`

And I think about again when should be output to stdout?

### When should be output to stdout?

1. without `--write`/`--check` flag
2. ~when `source content` not equal to `formatted content`~

**formatted code should always be printed to stdout.** How about it?

**Comment by kt3k** [comment]
@ry I think it's now safe to merge this PR. I checked `deno fmt` now maps to `run prettier/main.ts <files> --write`.

```
$ deno version
deno: 0.6.0
v8: 7.6.53
typescript: 3.4.1
$ deno fmt -D . | grep prettier/main | grep JS
DEBUG JS - args [ "https://deno.land/std/prettier/main.ts", ".", "--write" ]
```

File: prettier/main.ts [hunk 1]
```diff
@@ -139,15 +140,20 @@ async function formatFile(
...
   }
 
-  const formatted = prettier.format(text, {
+  const formatted: string = prettier.format(text, {
     ...prettierOpts,
     parser,
     plugins: prettierPlugins
   });
 
   if (text !== formatted) {
```
Comment (axetroy) [inline prettier/main.ts]: There is a question here that needs to be discussed.

If format a formatted file, do we still need to output to `stdout`?

Comment (kt3k) [reply prettier/main.ts]: The original prettier cli seems working like that (output always to stdout when `--write` is not specified). So I think it's better to follow it.

Comment (axetroy) [reply prettier/main.ts]: This may still have some problems, give me some time to confirm.

**Comment by axetroy** [comment]
waiting for #424

**Comment by ry** [comment]
Thanks @axetroy, @k3tk !

**Pull request merged**